### PR TITLE
Trigger cron when switching between RUCSS notices

### PR DESCRIPTION
**Pull request by remyperona**
## Description

Send a request to the cron when switching between RUCSS notices, to make sure the scheduled actions are running even if there is no activity on the website.

Fixes https://github.com/wp-media/nodejs-treeshaker/issues/29

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality)

## How Has This Been Tested?

- [x] Updated tests to validate the call

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: inc/Engine/Optimization/RUCSS/Admin/Subscriber.php [hunk 1]
```diff
@@ -581,5 +581,13 @@ private function set_notice_transient() {
...
 			MINUTE_IN_SECONDS
 		);
+
+		wp_safe_remote_get(
+			site_url( '/wp-cron.php' ),
+			[
+				'blocking' => false,
+				'timeout'  => 0.01,
+			]
+		);
```
Comment (engahmeds3ed) [inline inc/Engine/Optimization/RUCSS/Admin/Subscriber.php]: What about using this function instead?
https://developer.wordpress.org/reference/functions/spawn_cron/

it's doing the same functionality with additional checks/validations to make sure that we will not affect the performance.

**Review by engahmeds3ed — CHANGES_REQUESTED** [review]

File: inc/Engine/Optimization/RUCSS/Admin/Subscriber.php (hunk 1 above)
Comment (remyperona) [reply inc/Engine/Optimization/RUCSS/Admin/Subscriber.php]: Indeed it's much better, I didn't know about that one 👍🏼

**Review by engahmeds3ed — APPROVED** [review]

**Comment by Mai-Saad** [comment]
Checks:
- enable RUCSS while preload enabled/disabled=> both cases, the home page was still pending in used CSS table till user interaction, shouldn't it be completed with no need for user interaction on the dashboard? @Tabrisrp 
- clear used CSS while preload enabled /disabled => nothing added in db if preload disabled (will be handled in other PR), if preload was enabled, no entry added in the table for home unless if we preload cache (which is expected)
- edit safelist while preload enabled/disabled => while preload is enabled, entries are recreated but stuck in pending till DB interaction.. if preload is disabled, entry readded to home but stuck at pending (same as 1st point)

**Comment by remyperona** [comment]
I changed the approach to send the cron request at the moment we switch notices, it's ready to be tested again

**Pull request merged**